Modules/_blake2/blake2b_impl.c

```diff
@@ -334,12 +334,29 @@ _blake2_blake2b_hexdigest_impl(BLAKE2bObject *self)
     return _Py_strhex((const char *)digest, self->param.digest_length);
 }
 
+/*[clinic input]
+_blake2.blake2b.__reduce__
+
+Return the digest value as a string of hexadecimal digits.
+[clinic start generated code]*/
+
+static PyObject *
+_blake2_blake2b___reduce___impl(BLAKE2bObject *self)
+/*[clinic end generated code: output=3aeb49ad77e385ca input=6155b63cd93bf62c]*/
+{
+    PyErr_Format(PyExc_TypeError,
+                 "cannot pickle %s object",
+                 Py_TYPE(self)->tp_name);
```

Comment:
Rather than having to copy/paste such code in each type, would it be possible to have a protocol like:

```
class NotHashable:
    __hash__ = None

obj = NotHashable()
hash(obj)
```

This code raises `TypeError: unhashable type: 'NotHashable'`.

cc @pitrou @serhiy-storchaka